p2p/src/net/mock/backend.rs

```diff
@@ -407,6 +452,12 @@ where
                 command = self.cmd_rx.recv().fuse() => {
                     self.handle_command(command.ok_or(P2pError::ChannelClosed)?).await?;
                 }
+                ids = self.timeouts_receiver.recv() => {
+                    match ids {
+                        Some((request_id, peer_id)) => self.handle_request_timeout(peer_id, request_id).await?,
+                        None => return Err(P2pError::Other("Timeouts channels closed")),
```

Comment:
Since `timeouts_sender` is a part of `self`, I think this could be a panic.

Reply:
You are right, but I'm very reluctant to use panics in the code. 🙃 I can change it if you think it is OK.